### PR TITLE
Added support for specifying a TTL for redis keys (Fixes #1704)

### LINKED ISSUE #1704: Redis cache TTL
I've had a site with PageSpeed and Redis cache running for a few months now, and when running `redis-cli INFO` I've noticed this:

```
db1:keys=840116,expires=0,avg_ttl=0
```

It seems like PageSpeed doesn't set expire for the shared memory cache, keeping lots of unnecessary keys in the cache. Is there a way to set the expire time?

### PR DESCRIPTION
**Pull request by mtorromeo**
This is an attempt at implementing support for setting an expiration time to redis keys.
Issue #1704 describes the problem that this is trying to solve.

I have not developed in C++ in years and am not too familiar with the pagespeed codebase but I hope this is fine.

### CODE REVIEW

**Review by jmarantz — COMMENTED** [review]
Basically looks good except for some nits one significant omission. See https://github.com/apache/incubator-pagespeed-mod/blob/master/pagespeed/system/system_caches.cc#L288

The module keeps a map of RedisCache instances, which is keyed by a string composed of all the config options that are used to create the redis cache instance. You need to add your new TTL field into that string.

**Comment by mtorromeo** [comment]
I should have applied all the changes.
I also fixed a typo in the rewrite option name and an incorrect default of `0` for the ttl.

**Comment by oschaaf** [comment]
Awesome. It would be nice to have a system test for this, to make sure the feature doesn't get broken.
But I would be happy to merge this as-is. @mtorromeo wdyt?

File: pagespeed/system/system_rewrite_options.cc [hunk 1]
```diff
@@ -55,6 +56,8 @@ const char SystemRewriteOptions::kRedisReconnectionDelayMs[] =
 const char SystemRewriteOptions::kRedisTimeoutUs[] = "RedisTimeoutUs";
 const char SystemRewriteOptions::kRedisDatabaseIndex[] =
     "RedisDatabaseIndex";
+const char SystemRewriteOptions::kRedisTTLSec[] =
+    "RedisTTLSec";
```
Comment (jmarantz) [inline pagespeed/system/system_rewrite_options.cc]: fits on one line?

**Review by jmarantz — COMMENTED** [review]
You can, for no cost, get VirtualBox installed and an Ubuntu image and run tests on it.

Everything basically looks great but I agree with Otto it would be very nice to have tests for it.

**Comment by oschaaf** [comment]
I propose we merge this, but leave https://github.com/apache/incubator-pagespeed-mod/issues/1704 open and make a remark there that the feature is there but needs test coverage.

**Comment by jmarantz** [comment]
ok, sgtm.

**Pull request merged**